src/Controls/src/Core/Platform/AlertManager/AlertManager.Android.cs

```diff
@@ -94,6 +94,12 @@ public void ResetBusyCount()
 
 			void OnPageBusy(IView sender, bool enabled)
 			{
+				// Verify that the page making the request is part of this activity 
+				if (!PageIsInThisContext(sender))
```

Comment:
Can we add some tests for this one? 

You should be able to use `CreateHandlerAndAddToWindow` to open up multiple windows and set `IsBusy`

It's not completely accurate on android but it is on `Windows` and will be eventually accurate once we have these tests opening up new `Activities`

Reply:
Already there is a Device test like that here https://github.com/dotnet/maui/blob/4ac9f5bfacb9cbbac505615983b1538f5a448277/src/Controls/tests/DeviceTests/Elements/Page/PageTests.cs#L86